tests/modules/images/fetchpriority/fetchpriority-tests.php

```diff
@@ -0,0 +1,77 @@
+ *
+ * @covers ::fetchpriority_img_tag_add_attr
+ */
+class Fetchpriority_Test extends WP_UnitTestCase {
+	protected static $post;
+	protected static $attachment_id;
+	protected static $attachment_id_2;
+
+	protected $current_size_filter_data   = null;
+	protected $current_size_filter_result = null;
+
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		self::$post            = $factory->post->create_and_get();
+		$file                  = DIR_TESTDATA . '/images/canola.jpg';
+		self::$attachment_id   = $factory->attachment->create_upload_object(
+			$file,
+			self::$post->ID,
+			array(
+				'post_mime_type' => 'image/jpeg',
+			)
+		);
+		self::$attachment_id_2 = $factory->attachment->create_upload_object(
+			$file,
+			self::$post->ID,
+			array(
+				'post_mime_type' => 'image/jpeg',
+			)
+		);
+	}
+
+	public static function tear_down_after_class() {
+		wp_delete_attachment( self::$attachment_id, true );
+		wp_delete_attachment( self::$attachment_id_2, true );
+		parent::tear_down_after_class();
+	}
+
+	public function test_fetchpriority_img_tag_add_attr_based_on_context_and_loading_lazy() {
+		$img = get_image_tag( self::$attachment_id, '', '', '', 'large' );
+
+		$this->assertStringContainsString( 'fetchpriority="high"', fetchpriority_img_tag_add_attr( $img, 'the_content' ) );
+		$this->assertStringNotContainsString( 'fetchpriority="high"', fetchpriority_img_tag_add_attr( $img, 'not_content' ) );
+
+		$img = str_replace( '<img ', '<img loading="lazy" ', $img );
+		$this->assertStringNotContainsString( 'fetchpriority="high"', fetchpriority_img_tag_add_attr( $img, 'the_content' ) );
+	}
+
+	public function test_fetchpriority_img_tag_add_in_wp_filter_content_tags() {
+		global $wp_query;
+		global $wp_the_query;
+		$img   = get_image_tag( self::$attachment_id, '', '', '', 'large' );
+		$img_2 = get_image_tag( self::$attachment_id_2, '', '', '', 'large' );
+
+		$img = '<!-- wp:image {"id":' . self::$attachment_id . ',"sizeSlug":"large","linkDestination":"none"} -->
+<figure class="wp-block-image size-large">' . $img . '</figure>
+<!-- /wp:image -->
+<!-- wp:paragraph -->
+<p>This is an example page.</p>
+<!-- /wp:paragraph -->
+<!-- wp:image {"id":' . self::$attachment_id_2 . ',"sizeSlug":"large","linkDestination":"none"} -->
+<figure class="wp-block-image size-large">' . $img_2 . '</figure>
+<!-- /wp:image -->
+';
+		// Ensure image filtering occurs 'in_the_loop', is_main_query.
+		$wp_the_query          = $wp_query;
+		$wp_query->in_the_loop = true;
+		$content               = wp_filter_content_tags( $img, 'the_content' );
+		$this->assertStringContainsString( 'fetchpriority="high"', $content );
+		$this->assertStringContainsString( 'loading="lazy"', $content );
+		$this->assertTrue( strpos( $content, 'fetchpriority="high"' ) < strpos( $content, 'loading="lazy"' ) );
+
+		$spilt = explode( 'fetchpriority', $content );
+		$this->assertEquals( 2, count( $spilt ) );
```

Comment:
What is the purpose of this? To check how often `fetchpriority` is included in the content? In that case we should rather use `substr_count`. This logic looks confusing :)

Reply:
i was confused at first as well, this is just counting that fetchpriority is included exactly once, agree changing this to a substr_count makes more sense 👍🏼

Reply:
Adjusted